src/hypothesis/internal/conjecture/utils.py

```diff
@@ -71,8 +71,26 @@ def centered_integer_range(data, lower, upper, center):
     )
 
 
+def is_deprecated_sample_type(values):
+    try:
+        from numpy import ndarray
+        if isinstance(values, ndarray):
+            # This check is used to determine if `values` is a
+            # suitable argument for
+            # `hypothesis.strategies.sample_from`. That method returns
+            # a single element (e.g., an integer). We check for
+            # one-dimensional array here because any multi-dimensional
+            # arrays would return an array (not an element from that
+            # array as expected).
+            return values.ndim != 1
+    except ImportError:  # pragma: no cover
+        pass
+
+    return not isinstance(values, (OrderedDict, Sequence, enum.EnumMeta))
+
+
 def check_sample(values):
-    if not isinstance(values, (OrderedDict, Sequence, enum.EnumMeta)):
+    if is_deprecated_sample_type(values):
```

Comment:
I can see why you split it out, but I'm going to be annoying and ask that you inline it - so that we can give a distinct error message for >1-dimensional arrays.  Eg:

```python
def check_sample(values):
    try:
        from numpy import ndarray
        if isinstance(values, ndarray):
            if values.ndim != 1:
                note_deprecation((
                    'Only one-dimensional arrays are supported for sampling, '
                    'and the given value has {ndim} dimensions (shape '
                    '{shape}).  This array will give samples of array slices '
                    'instead of elements!  Use np.ravel(values) to convert '
                    'to a one-dimensional array, or tuple(values) if you '
                    'want to sample slices.  This will be an error in a '
                    'future version of Hypothesis.').format(
                        ndim=values.ndim, shape=values.shape))
            return tuple(values)
    except ImportError:  # pragma: no cover
        pass
    if not isinstance(values, (OrderedDict, Sequence, enum.EnumMeta)):
        note_deprecation(
            ('Cannot sample from %r, not a sequence.  ' % (values,)) +
            'Hypothesis goes to some length to ensure that sampling an '
            'element from a collection (with `sampled_from` or `choices`) is '
            'replayable and can be minimised.  To replay a saved example, '
            'the sampled values must have the same iteration order on every '
            'run - ruling out sets, dicts, etc due to hash randomisation.  '
            'Most cases can simply use `sorted(values)`, but mixed types or '
            'special values such as math.nan require careful handling - and '
            'note that when simplifying an example, Hypothesis treats '
            'earlier values as simpler.')
    return tuple(values)
```